### ISSUE TITLE
Add lightweight unicode plot to Histogram show for text displays?

### ISSUE THREAD

**Issue by oschulz** [JuliaStats/StatsBase.jl#786]
Something like this

```julia
function Base.show(io::IO, ::MIME"text/plain", h::Histogram{<:Real,1})
    show(io, h)
    println(io)
    edge = first(h.edges)
    if edge isa AbstractRange && length(eachindex(edge)) <= 120
        W = h.weights
        barsyms = [' ', '▁', '▂', '▃', '▄', '▅', '▆', '▇', '█']
        symidxs = eachindex(barsyms)
        norm_factor = length(symidxs) / maximum(W)
        get_sym_idx(x) = isnan(x) ? 1 : clamp(first(symidxs) + floor(Int, norm_factor * x), first(symidxs), last(symidxs))
        print(io, minimum(edge))
        print(io, h.closed == :left ? "[" : "]")
        print(io, String(barsyms[get_sym_idx.(W)]))
        print(io, h.closed == :right ? "]" : "[")
        print(io, maximum(edge))   
    end
end
```

would get us a neat, lightweight display for 1D-histograms:

```julia
julia> h = fit(Histogram, randn(10^4), nbins = 80)
Histogram{Int64, 1, Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}}
edges:
  -3.9:0.1:4.3
weights: [1, 0, 0, 1, 0, 1, 3, 1, 5, 7  …  2, 2, 3, 2, 0, 0, 1, 0, 0, 1]
closed: left
isdensity: false
-3.9[                 ▁▁▁▁▁▂▂▃▃▄▄▅▅▆▆▆▆█▇▇█████▇█▇▇▆▅▅▅▄▃▃▃▃▂▂▁▁▁                      [4.3
```

Instead of limiting to 120 bins like above we could rebin "long" histograms.

If acceptable, I would do a PR.

**Comment by pdeffebach** [comment]
I'm always skeptical of statistics libraries that also incorporate plotting. 

This seems like it might be annoying to have as an output (I'm not a fan of the new `@benchmark` histograms). Given that UnicodePlots.jl is such a lightweight dependency, I'm not sure this is needed.

**Comment by oschulz** [comment]
> UnicodePlots.jl is such a lightweight dependency

I would say this is debatable. :-) Unicode currently has 90 dependencies (direct and indirect) and a load time of about 2 seconds.

**Comment by pdeffebach** [comment]
Well, even more than that, StatsBase.jl is a dependency of unicode plots! 

Fair enough. I still think the histograms are a cluttered. Maybe we can add a summary method which prints the output? Maybe quantiles.

**Comment by oschulz** [comment]
I guess it's a matter of preference - packages like Colors.jl do a graphical display as well (only in HTML, in that case) after all. I would enjoy seeing 1D-histograms "directly", but I can understand others might prefer text-only.

@nalimilan , what's your take on this?

**Comment by oschulz** [comment]
CC @andreasnoack

**Comment by nalimilan** [comment]
I'm hesitant. This looks nice, but I wonder whether it's really useful given that there's no indication of the ticks on the x axis.

**Comment by oschulz** [comment]
> I wonder whether it's really useful given that there's no indication of the ticks on the x axis

True, it wouldn't be a plot that gives detailed information. But it would, I think, provide more information at a glance than the current purely text-based summary, and could even be more compact.
